test-suite/js/src/test/require-modules/org/scalajs/testsuite/jsinterop/ModulesTest.scala

```diff
@@ -0,0 +1,113 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2016, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.jsinterop
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+
+import org.junit.Assert._
+import org.junit.Test
+
+class ModulesTest {
+  import ModulesTest._
+
+  @Test def testImportModuleItself(): Unit = {
+    val qs = QueryString
+    assertTrue(qs.isInstanceOf[js.Object])
+
+    val dict = js.Dictionary("foo" -> "bar", "baz" -> "qux")
+
+    assertEquals("foo=bar&baz=qux", qs.stringify(dict))
+    assertEquals("foo:bar;baz:qux", qs.stringify(dict, ";", ":"))
+
+    /* Potentially, this could be "optimized" by importing `stringify` as a
+     * global symbol if we are emitting ES2015 modules.
+     */
+    assertEquals("foo=bar&baz=qux", QueryString.stringify(dict))
+    assertEquals("foo:bar;baz:qux", QueryString.stringify(dict, ";", ":"))
+  }
+
+  @Test def testImportLegacyModuleItselfAsDefault(): Unit = {
+    val qs = QueryStringAsDefault
+    assertTrue(qs.isInstanceOf[js.Object])
+
+    val dict = js.Dictionary("foo" -> "bar", "baz" -> "qux")
+
+    assertEquals("foo=bar&baz=qux", qs.stringify(dict))
+    assertEquals("foo:bar;baz:qux", qs.stringify(dict, ";", ":"))
+
+    /* Potentially, this could be "optimized" by importing `stringify` as a
+     * global symbol if we are emitting ES2015 modules.
+     */
+    assertEquals("foo=bar&baz=qux", QueryStringAsDefault.stringify(dict))
+    assertEquals("foo:bar;baz:qux", QueryStringAsDefault.stringify(dict, ";", ":"))
+  }
+
+  @Test def testImportObjectInModule(): Unit = {
+    assertTrue((Buffer: Any).isInstanceOf[js.Object])
+    assertFalse(Buffer.isBuffer(5))
+  }
+
+  @Test def testImportClassInModule(): Unit = {
+    val b = new Buffer(5)
+    for (i <- 0 until 5)
+      b(i) = (i * i).toShort
+
+    for (i <- 0 until 5)
+      assertEquals(i * i, b(i).toInt)
```

Comment:
That would make `testImportClassInModule` dependent on what is supposed to be tested by `testImportObjectInModule`.

However I could definitely add that in `testImportIntegrated`.